sdk/core/core-client-rest/src/multipart.ts

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  BodyPart,
+  MultipartRequestBody,
+  RestError,
+  createHttpHeaders,
+} from "@azure/core-rest-pipeline";
+import { stringToUint8Array } from "@azure/core-util";
+import { isBinaryBody } from "./helpers/isBinaryBody.js";
+
+/**
+ * Describes a single part in a multipart body.
+ */
+export interface PartDescriptor {
+  /**
+   * Content type of this part. If set, this value will be used to set the Content-Type MIME header for this part, although explicitly
+   * setting the Content-Type header in the headers bag will override this value. If set to `null`, no content type will be inferred from
+   * the body field. Otherwise, the value of the Content-Type MIME header will be inferred based on the type of the body.
+   */
+  contentType?: string | null;
+
+  /**
+   * The disposition type of this part (for example, "form-data" for parts making up a multipart/form-data request). If set, this value
+   * will be used to set the Content-Disposition MIME header for this part, in addition to the `name` and `filename` properties.
+   * Explicitly setting the Content-Disposition header in the headers bag will override this value.
+   */
+  dispositionType?: string;
+
+  /**
+   * The field name associated with this part. This value will be appended to the Content-Disposition header if dispositionType is set.
+   */
+  name?: string;
```

Comment:
just curious will the field name be optional?

Reply:
At the Core level, it's optional, so we can support other multipart content types, but in the generated models for MFD it will be required.